src/provider.ts

```diff
@@ -150,12 +147,15 @@ export function JsonRpcApiProvider<TBase extends Constructor<ethers.JsonRpcApiPr
             }
         }
 
-        async l2TokenAddress(token: Address): Promise<string> {
+        async l2TokenAddress(token: Address): Promise<Address> {
             if (token == ETH_ADDRESS) {
                 return ETH_ADDRESS;
+            } else if (await this.isBaseTokenAddress(token)) {
+                // baseToken doesn't have a token address
+                return "0x";
```

Comment:
the base token address is actually `0x000000000000000000000000000000000000800a` = ETH_ADDRESS

Reply:
fixed